vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL),
+  filter = PARAMCD == "ICB"
+)
+```
+
+## Derive Best Overall Response Parameter {#bor}
+
+The function `admiral::derive_extreme_event()` can be used to derive the best
+overall response (without confirmation required) parameter. Similar to the above
+function you can optionally decide what period would you consider an `iSD` or
+`NON-iCR/NON-iUPD` as being eligible from. In this example, 42 days after
+randomization date has been used again.
+
+Please note that the order of the events specified for `events` is important.
+For example, a subject with `iPR`, `iPR`, `iCR` qualifies for both `ibor_icr` and
+`ibor_ipr`. As `ibor_icr` is listed before `ibor_ipr`, `iCR` is selected as best overall
+response for this subject.
+
+```{r}
+ibor_icr <- event(
+  description = "Define complete response (iCR) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iCR",
+  set_values_to = exprs(AVALC = "iCR")
+)
+
+ibor_ipr <- event(
+  description = "Define partial response (iPR) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iPR",
+  set_values_to = exprs(AVALC = "iPR")
+)
+
+ibor_isd <- event(
+  description = paste(
+    "Define stable disease (iSD) for best overall response (iBOR) as iCR, iPR, or iSD",
+    "occurring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iCR", "iPR", "iSD") & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "iSD")
+)
+
+ibor_non_icripd <- event(
+  description = paste(
+    "Define NON-iCR/NON-iUPD for best overall response (iBOR) as NON-iCR/NON-iUPD",
+    "occuring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC == "NON-iCR/NON-iUPD" & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "NON-iCR/NON-iUPD")
+)
+
+ibor_icpd <- event(
+  description = "Define progressive disease (iCPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iCPD",
+  set_values_to = exprs(AVALC = "iCPD")
+)
+
+ibor_iupd <- event(
+  description = "Define progressive disease (iUPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iUPD",
+  set_values_to = exprs(AVALC = "iUPD")
+)
+
+ibor_ne <- event(
+  description = paste(
+    "Define not evaluable (NE) for best overall response (iBOR) as iSD, NON-iCR/NON-iUPD,",
+    "or NE (should be specified after ibor_isd and ibor_non_icripd)"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iSD", "NON-iCR/NON-iUPD", "NE"),
+  set_values_to = exprs(AVALC = "NE")
+)
+
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(ibor_icr, ibor_ipr, ibor_isd, ibor_non_icripd, ibor_icpd, ibor_iupd, ibor_ne, no_data_missing),
```

Comment:
It seems to me that we should not differentiate the progression between iUPD and iCPD. Instead, a single iPD value with the date when the iUPD was first identified. See slides 66, 69 and 72

Reply:
@starosto I am not sure why the slides #66 mentioned PD as iBOR, we don't have PD as iRECIST response category. @vinhn23 @bundfussr For iBOR( unconfirmed) we need to think as this event list which use min. date value and after pick iUPD over iCPD.
Based on my discussion internally on this :  we should only be assigning iUPD as BOR when a) no better response assessment exists, and b) when iCPD does not subsequently occur. When (any number of) iUPD TPRs are followed by iCPD, we should be considering them as a single block of confirmed PD for the purposes of deriving BOR and date of PD. 
I checked the iBOR for 701-1028 , iBOR looks as I expected..

Reply:
I took part in a study with PCWG3 criteria (Bone scans + Recist 1.1). In the bone scans uPD (unconfirmed progression) was also defined, but in the instructions it was written that as BOR we do not differentiate between uPD and PD and the Investigator filling out the BOR form (RSTESTCD=BESTRESP) is obliged to change uPD to PD.

On the other hand, there is an article on the EORTC website https://recist.eortc.org/recist/wp-content/uploads/sites/4/2017/03/Manuscript_IRECIST_Lancet-Oncology_Seymour-et-al_revision_FINAL_clean_nov25.pdf and on page 29 it shows iUPD and iCPD as iBOR categories.

I will investigate internally what BI's expectations would be regarding progression and iBOR.

For sure the date of progression needs to be corrected with iCPD BOR and I still wonder about PARAMCD=ICPD, if it is sufficient, because it only indicates two patients with progression, while patients 01-701-1015 or 01-701-1115 are also suspicious.

Reply:
will update with new code